src/main/java/com/google/sps/data/User.java

```diff
@@ -1,57 +1,198 @@
 package com.google.sps.data;
 
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.lang.reflect.Type;
 import java.util.ArrayList;
 import java.util.List;
 
-// Class representing a User
 public class User {
-  private ArrayList<Long> cookbook;
-  private ArrayList<Long> planner;
-  private ArrayList<Long> userRecipes;
-  private String email;
-  private String id;
-  private String username;
-  // shopping list will not be implented until after MVP
-  private ArrayList<String> shoppingList;
+  // API variables:
+  // client ID is generated from Ali's google APIs credentials page.
+  private static String CLIENT_ID = "1034390229233-u07o0iaas2oql8l4jhe7fevpfsbrtv7n.apps.googleusercontent.com";
+  // Payload class contains decrypted user information.
+  private Payload payload;
+  private Entity entity;
 
-  public User(String id, String email, String username) {
-    this.id = id;
-    this.email = email;
-    this.username = username;
-    planner = new ArrayList<Long>();
-    cookbook = new ArrayList<Long>();
-    userRecipes = new ArrayList<Long>();
+  /**
+    // User Entity Properties:
+
+    "displayName" : "Example Name"
+    "cookbook" : [ "longId1", "longId2", "longId3" ]
+    "userRecipes" : [ "longId1", "longId2", "longId3" ]
+    "planner" : [ "longId1", "longId2", "longId3" ]
+    "shoppingList" : [ "longId1", "longId2", "longId3" ]
+
+    // Properties that can be accessed from Payload object:
+
+    // These seven fields are only included when the user has granted the "profile" and
+    // "email" OAuth scopes to the application.
+    // "sub" or Subject refers to the user-unique ID number.
+    "sub" : "110169484474386276334"
+    "email" : "testuser@gmail.com",
+    "email_verified" : "true",
+    "name" : "Test User",
+    "picture" : "https://lh4.googleusercontent.com/-kYgzyAWpZzJ/ABCDEFGHI/AAAJKLMNOP/tIXL9Ir44LE/s99-c/photo.jpg",
+    "given_name" : "Test",
+    "family_name" : "User",
+  */
```

Comment:
A couple things about this comment block:
1) Since you have a comment block with `/* ... */` you can remove all the`//` within the block
2) It seems like it would be more helpful to put the comments above the variables they relate to, like:
`/* comment about payload */`
`private Payload payload;`
`/* comment about entity */`
`private Entity entity;`
3) Are the long IDs stored as strings? It seems confusing to be representing them as "longId1", etc when you would be expecting a number, ie 1 or "1" (depending on the answer to the first question)